src/openrct2/EditorObjectSelectionSession.cpp

```diff
@@ -321,7 +321,7 @@ void unload_unselected_objects()
     const ObjectRepositoryItem* items = object_repository_get_items();
 
     size_t numObjectsToUnload = 0;
-    rct_object_entry* objectsToUnload = static_cast<rct_object_entry*>(malloc(numItems * sizeof(rct_object_entry)));
+    std::unique_ptr<rct_object_entry[]> objectsToUnload = std::make_unique<rct_object_entry[]>(numItems);
```

Comment:
This can be changed to a std::vector. In a future refactor object_manager_unload_objects can take the vector.

Reply:
Yeah I know I said it could be used as a unique ptr but was reading more about it online and it makes sense to change it to a vector.

Reply:
@duncanspumpkin Okay then. In that case, should I modify object_manager_unload_objects() to take a vector as its first argument?

Reply:
Depends if it looks like that requires a lot of work or not. We like to keep the PRs small and manageable.

Reply:
@duncanspumpkin It's not bad at all. Only two functions use it (both in the ObjectManager class: object_manager_unload_objects and UnloadObjects).